### PR TITLE
Add signal to let plugins register additional fonts for designer templates.

### PR DESCRIPTION
**Pull request by vasantvohra**
Under discussion...

### CODE REVIEW

File: indico/modules/designer/templates/template.html [hunk 1]
```diff
@@ -305,7 +304,7 @@
                 <div class="flexrow f-a-baseline">
                     <label for="element-list">{% trans %}Insert element:{% endtrans %}</label>
                     <select id="element-list">
-                        {% for group_id, group in placeholders.items() | sort(attribute='1.title') %}
+                        {% for group_id, group in placeholders.viewitems() | sort(attribute='1.title') %}
```
Comment (ThiefMaster) [inline indico/modules/designer/templates/template.html:307]: That's Python-2-only...

File: indico/legacy/pdfinterface/base.py [hunk 2]
```diff
@@ -134,6 +135,7 @@ def setTTFonts():
         addMapping('LinuxLibertine', 1, 1, 'LinuxLibertine-Bold-Italic')
         pdfmetrics.registerFont(TTFont('Kochi-Mincho', os.path.join(font_dir, 'kochi-mincho-subst.ttf')))
         pdfmetrics.registerFont(TTFont('Kochi-Gothic', os.path.join(font_dir, 'kochi-gothic-subst.ttf')))
+        signals.legacy.register_additional_fonts.send()
```
Comment (ThiefMaster) [inline indico/legacy/pdfinterface/base.py:138]: I'd rather have the signal return the font specs (name and path?) and do the registration here...

File: indico/modules/designer/templates/template.html (hunk 1 above)
Comment (vasantvohra) [reply indico/modules/designer/templates/template.html]: Yes I was in the process of reverting those lines

Comment (vasantvohra) [reply indico/modules/designer/templates/template.html]: What about two separate dictionaries? is it ok?

File: indico/legacy/pdfinterface/base.py (hunk 2 above)
Comment (vasantvohra) [reply indico/legacy/pdfinterface/base.py:138]: Okay, but then how to go ahead with `addMapping`?

Comment (vasantvohra) [reply indico/legacy/pdfinterface/base.py:138]: @ThiefMaster,

-  I see `pdfmetrics.registerFont` calls `registerFontFamily(font.fontName)` so there's no need to explicitly do `addMapping` `from reportlab.lib.fonts import addMapping` ?

- Also, to return a name and path, do I just return a dictionary {font_name: path} from the function connected to the signal?

```python
plugin.connect(signals.legacy.register_additional_fonts, setAdditionalTTFonts)

def setAdditionalTTFonts(self):
    font_dir = os.path.join('../un/', 'indico_fonts') 
    return {'Bolt': os.path.join(font_dir, 'Bolt-Regular.ttf'), 'Bolt-Bold': os.path.join(font_dir, 'Bolt-Bold.ttf')}
```

Comment (ThiefMaster) [reply indico/legacy/pdfinterface/base.py:138]: tbh, this is legacy code, I don't think anyone in the team remembers the details why these things were done that way :)

**Comment by ThiefMaster** [comment]
closing in favor of #5003 to avoid adding new signals for legacy code

**Pull request closed without merging**